src/components/stepforms/FilterStepForm.vue

```diff
@@ -74,32 +78,26 @@ export default class FilterStepForm extends BaseStepForm<FilterStep> {
     }
   }
 
-  get defaultCondition() {
-    const cond: FilterSimpleCondition = { column: '', value: '', operator: 'eq' };
-    return cond;
-  }
-
-  get conditions() {
-    if (isFilterComboAnd(this.editedStep.condition) && this.editedStep.condition.and.length) {
-      return this.editedStep.condition.and;
+  get dataPath() {
+    if (isFilterCombo(this.editedStep.condition)) {
+      return Object.keys(this.editedStep.condition)[0];
     } else {
-      return [this.defaultCondition];
+      return '';
     }
   }
 
-  set conditions(newval) {
-    if (isFilterComboAnd(this.editedStep.condition)) {
-      this.editedStep.condition.and = [...newval];
-    }
+  get defaultCondition() {
+    const cond: FilterSimpleCondition = { column: '', value: '', operator: 'eq' };
+    return cond;
   }
 
-  get multipleConditionsClass() {
-    return {
-      'filter-form--multiple-conditions': this.conditions.length > 1,
-    };
+  get conditionsTree() {
+    return buildConditionsEditorTree(this.editedStep.condition);
   }
 
   submit() {
+    // FIXME !!!!
+    // This "if" works only for comboAnd, not for comboOr and simpleCondition, and not recursive for groups.
     if (isFilterComboAnd(this.editedStep.condition)) {
```

Comment:
Is it possible to do the fix now ? Why not ?

Reply:
Yeah, I just ran out of time...